make/cluster.sh

```diff
@@ -104,18 +104,15 @@ if printenv K8S_VERSION >/dev/null && [ -n "$K8S_VERSION" ]; then
   k8s_version="$K8S_VERSION"
 fi
 
-# TODO(SgtCoDFish): Using "KIND_IMAGE_FULL_*" here breaks podman, which doesn't support using both tags and digests
-# when referring to an image. We should avoid using FULL where possible.
```

Comment:
@SgtCoDFish How were you using `podman` when  you wrote this comment?  Is it easy to test it with `podman` again and see if these changes solve the problem?

Reply:
I was using `make CTR=podman ...` in an effort to reduce our reliance on docker. I can't really test easily any more because my setup on macOS is completely different to my linux setup

I'd love to not have a hard dependency on docker, but at the end of the day if we don't have automated tests for podman I don't think we support podman

Reply:
I tried using Kind + Podman as follows, but it still doesn't work.


```sh
$ export KIND_EXPERIMENTAL_PROVIDER=podman K8S_VERSION=1.27 CI=yes; make CTR=podman e2e-setup
curl --silent --show-error --fail --location --retry 10 --retry-connrefused https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.0.0/experimental-install.yaml -o _bin/downloaded/gateway-api-v1.0.0.yaml
./hack/util/checkhash.sh _bin/downloaded/gateway-api-v1.0.0.yaml 6c601dced7872a940d76fa667ae126ba718cb4c6db970d0bab49128ecc1192a3
GOBIN=/home/richard/projects/cert-manager/cert-manager/_bin/downloaded/tools/ go install github.com/google/go-containerregistry/cmd/crane@v0.16.1
cd _bin/tools/ && cp -f -r ../downloaded/tools/crane@v0.16.1_linux_amd64 crane
/home/richard/projects/cert-manager/cert-manager/_bin/tools/crane pull docker.io/kindest/node@sha256:3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72 _bin/downloaded/containers/amd64/docker.io/kindest/node@sha256+3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72.tar --platform linux/amd64
podman inspect docker.io/kindest/node@sha256:3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72 2>/dev/null >&2 || podman load -i _bin/downloaded/containers/amd64/docker.io/kindest/node@sha256+3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72.tar
WARN[0000] The cgroupv2 manager is set to systemd but there is no systemd user session available
WARN[0000] For using systemd, you may need to login using an user session
WARN[0000] Alternatively, you can enable lingering with: `loginctl enable-linger 1000` (possibly as root)
WARN[0000] Falling back to --cgroup-manager=cgroupfs
WARN[0000] Error validating CNI config file /home/richard/.config/cni/net.d/kind.conflist: [plugin bridge does not support config version "1.0.0" plugin portmap does not support config version "1.0.0" plugin firewall does not support config version "1.0.0" plugin tuning does not support config version "1.0.0"]
Getting image source signatures
Copying blob bf735ea575cf done
Copying blob 864b5c27b966 done
Copying config 89e7dc9f91 done
Writing manifest to image destination
Storing signatures
Loaded image(s): docker.io/kindest/node:i-was-a-digest
curl --silent --show-error --fail --location --retry 10 --retry-connrefused https://github.com/kubernetes-sigs/kind/releases/download/v0.20.0/kind-linux-amd64 -o _bin/downloaded/tools/kind@v0.20.0_linux_amd64
./hack/util/checkhash.sh _bin/downloaded/tools/kind@v0.20.0_linux_amd64 513a7213d6d3332dd9ef27c24dab35e5ef10a04fa27274fe1c14d8a246493ded
chmod +x _bin/downloaded/tools/kind@v0.20.0_linux_amd64
cd _bin/tools/ && cp -f -r ../downloaded/tools/kind@v0.20.0_linux_amd64 kind
curl --silent --show-error --fail --location --retry 10 --retry-connrefused https://storage.googleapis.com/kubernetes-release/release/v1.28.1/bin/linux/amd64/kubectl -o _bin/downloaded/tools/kubectl@v1.28.1_linux_amd64
./hack/util/checkhash.sh _bin/downloaded/tools/kubectl@v1.28.1_linux_amd64 e7a7d6f9d06fab38b4128785aa80f65c54f6675a0d2abef655259ddd852274e1
chmod +x _bin/downloaded/tools/kubectl@v1.28.1_linux_amd64
cd _bin/tools/ && cp -f -r ../downloaded/tools/kubectl@v1.28.1_linux_amd64 kubectl
curl --silent --show-error --fail --location --retry 10 --retry-connrefused https://github.com/mikefarah/yq/releases/download/v4.35.1/yq_linux_amd64 -o _bin/downloaded/tools/yq@v4.35.1_linux_amd64
./hack/util/checkhash.sh _bin/downloaded/tools/yq@v4.35.1_linux_amd64 bd695a6513f1196aeda17b174a15e9c351843fb1cef5f9be0af170f2dd744f08
chmod +x _bin/downloaded/tools/yq@v4.35.1_linux_amd64
cd _bin/tools/ && cp -f -r ../downloaded/tools/yq@v4.35.1_linux_amd64 yq
kind create cluster --config make/config/kind/cluster.yaml --image docker.io/kindest/node@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31 --name kind
using podman due to KIND_EXPERIMENTAL_PROVIDER
enabling experimental podman provider
Cgroup controller detection is not implemented for Podman. If you see cgroup-related errors, you might need to set systemd property "Delegate=yes", see https://kind.sigs.k8s.io/docs/user/rootless/
Creating cluster "kind" ...
 ✓ Ensuring node image (docker.io/kindest/node) 🖼
 ✗ Preparing nodes 📦
Deleted nodes: ["kind-control-plane"]
ERROR: failed to create cluster: command "podman run --name kind-control-plane --hostname kind-control-plane --label io.x-k8s.kind.role=control-plane --privileged --tmpfs /tmp --tmpfs /run --volume 71444fa0cc37ddde20139db88c2cebd5eeea61b07ecd9a17b19db8969cdf643e:/var:suid,exec,dev --volume /lib/modules:/lib/modules:ro -e KIND_EXPERIMENTAL_CONTAINERD_SNAPSHOTTER --detach --tty --net kind --label io.x-k8s.kind.cluster=kind -e container=podman --cgroupns=private --device /dev/fuse --publish=127.0.0.1:43175:6443/tcp -e KUBECONFIG=/etc/kubernetes/admin.conf docker.io/kindest/node@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31" failed with error: exit status 127
Command Output: time="2024-01-26T11:17:11Z" level=warning msg="The cgroupv2 manager is set to systemd but there is no systemd user session available"
time="2024-01-26T11:17:11Z" level=warning msg="For using systemd, you may need to login using an user session"
time="2024-01-26T11:17:11Z" level=warning msg="Alternatively, you can enable lingering with: `loginctl enable-linger 1000` (possibly as root)"
time="2024-01-26T11:17:11Z" level=warning msg="Falling back to --cgroup-manager=cgroupfs"
time="2024-01-26T11:17:11Z" level=warning msg="Error validating CNI config file /home/richard/.config/cni/net.d/kind.conflist: [plugin bridge does not support config version \"1.0.0\" plugin portmap does not support config version \"1.0.0\" plugin firewall does not support config version \"1.0.0\" plugin tuning does not support config version \"1.0.0\"]"
time="2024-01-26T11:17:11Z" level=warning msg="Error validating CNI config file /home/richard/.config/cni/net.d/kind.conflist: [plugin bridge does not support config version \"1.0.0\" plugin portmap does not support config version \"1.0.0\" plugin firewall does not support config version \"1.0.0\" plugin tuning does not support config version \"1.0.0\"]"
time="2024-01-26T11:17:11Z" level=error msg="error loading cached network config: network \"kind\" not found in CNI cache"
time="2024-01-26T11:17:11Z" level=warning msg="falling back to loading from existing plugins on disk"
time="2024-01-26T11:17:11Z" level=warning msg="Error validating CNI config file /home/richard/.config/cni/net.d/kind.conflist: [plugin bridge does not support config version \"1.0.0\" plugin portmap does not support config version \"1.0.0\" plugin firewall does not support config version \"1.0.0\" plugin tuning does not support config version \"1.0.0\"]"
time="2024-01-26T11:17:11Z" level=error msg="Error tearing down partially created network namespace for container 16c93cdcb71693f729779134015c5f16348c1353894cf276c29a605729a2800c: CNI network \"kind\" not found"
Error: error configuring network namespace for container 16c93cdcb71693f729779134015c5f16348c1353894cf276c29a605729a2800c: CNI network "kind" not found
make: *** [make/e2e-setup.mk:88: _bin/scratch/kind-exists] Error 1
```

Reply:
Yeah ultimately if we don't test with podman I don't think we should bend over backwards making changes to support it. I think we can merge this without any worries!